generate_data_deterministic.py

```diff
@@ -0,0 +1,554 @@
+                                                  data=np.random.rand(num_rows_initial),
+                                                  dtype=(np.dtype('float64')),
+                                                  chunks=chunk_size,
+                                                  compression=compression)
+            else:
+                key0_ds = f.create_dataset(name + '/key0',
+                                           data=np.random.rand(num_rows_initial),
+                                           dtype=(np.dtype('int64')),
+                                           maxshape=(None,),
+                                           chunks=(chunk_size,),
+                                           compression=compression)
+                key1_ds = f.create_dataset(name + '/key1',
+                                           data=np.random.rand(num_rows_initial),
+                                           dtype=(np.dtype('int64')),
+                                           maxshape=(None,),
+                                           chunks=(chunk_size,),
+                                           compression=compression)
+                val_ds = f.create_dataset(name + '/val',
+                                          data=np.random.rand(num_rows_initial),
+                                          dtype=(np.dtype('float64')),
+                                          maxshape=(None,),
+                                          chunks=(chunk_size,),
+                                          compression=compression)
+
+            for a in range(num_transactions):
+                if print_transactions:
+                    print("Transaction", a)
+                tt = datetime.datetime.utcnow()
+                if versions:
+                    with file.stage_version(str(tt)) as group:
+                        key0_ds = group[name + '/key0']
+                        key1_ds = group[name + '/key1']
+                        val_ds = group[name + '/val']
+                        cls._modify_dss_sparse_deterministic(key0_ds,
+                                                             key1_ds,
+                                                             val_ds,
+                                                             num_rows_per_append,
+                                                             num_changes,
+                                                             num_deletes,
+                                                             num_inserts)
+                else:
+                    cls._modify_dss_sparse_deterministic(key0_ds, key1_ds, val_ds,
+                                                         num_rows_per_append,
+                                                         num_changes,
+                                                         num_deletes,
+                                                         num_inserts)
+
+                t = time.time()
+                times.append(t-told)
```

Comment:
It might remove some noise if you call `f.flush()` after each loop. Otherwise h5py may randomly decide to flush in the middle of some operations, which would cause the time to spike for that run.